ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -57,10 +58,10 @@ public MainnetTransactionValidator(
   public MainnetTransactionValidator(
       final GasCalculator gasCalculator,
       final boolean checkSignatureMalleability,
-      final int chainId) {
+      final BigInteger chainId) {
     this.gasCalculator = gasCalculator;
     this.disallowSignatureMalleability = checkSignatureMalleability;
-    this.chainId = chainId > 0 ? OptionalInt.of(chainId) : OptionalInt.empty();
+    this.chainId = chainId.signum() == 1 ? Optional.of(chainId) : Optional.empty();
```

Comment:
Just pass in `Optional<BigInteger>` and avoid the sentinel check here too.

Reply:
done